content/_layouts/docs.njk

```diff
@@ -7,22 +7,34 @@ layout: default
 </div>
 
 <div id="wrap" class="container" role="document">
-  <div id="responsive-menu"><button class='btn btn-primary' onclick='handleMenu()'>Table of Contents</button></div>
+  <div id="responsive-menu"><button class="btn btn-primary" onclick="handleMenu()">Table of Contents</button></div>
   <div id="content" class="row-fluid">
     <div class="toc well span3 visible-toc">
+      <div style="display: flex; flex-direction: row; align-items: space-between; justify-content: space-between;">
+        <input type="text" id="search-input" placeholder="Search docs" /><br/>        
+        <button style="width: 40% margin: auto; margin-left: 5px;" class="btn-primary" onclick="handleSearch()">Search</button>
+      </div>
       {% include "docs-toc.html" %}
     </div>
-    <div class="well span9 docs">
+    <div id="main-docs" class="well span9 docs">
       <h1>{{ title }}</h1>
-      {{ content | safe }}
+      {{ content | safe }}   
     </div>
   </div>
 </div>
 
 <script>
-    function handleMenu() {
-        const menu = document.querySelector('.toc');
-        menu.classList.toggle('hidden-toc');
-        menu.classList.toggle('visible-toc');
+    document.addEventListener("DOMContentLoaded", async () => {
+      const search = document.getElementById("search-input")
+        search.addEventListener("keyup", ({key}) => {
+          if (key === "Enter") {
+            handleSearch();
+          }
+        })
+    })
+    const handleSearch = async () => {
+      const searchTerm = document.getElementById("search-input").value;
+      const queryString = encodeURI(searchTerm)
+      window.location.replace(`/search?term=${queryString}`);
```

Comment:
Something about the way location is being set here is messing with the browser history. I was on a specific docs page, I did a search and then hit the back button and I didn't go back to the page that I was on. Not sure exactly what is going on, but I think you need to just be setting `window.location` and not doing a replace maybe? Basically, I just want to be on a random page, do a search, hit enter, view the results and then hit the back button and go back to where I was.